tests/kernel/run_ltp.pm

```diff
@@ -0,0 +1,319 @@
+        if ($line =~ /(^#)|(^$)/) {
+            next;
+        }
+
+        #Command format is "<name> <command> [<args>...] [#<comment>]"
+        if ($line =~ /^\s* (\w+) \s+ (\S.+) #?/gx) {
+            my $test = {name => $1, command => $2};
+            if ($test->{name} =~ $cmd_pattern) {
+                push @tests, $test;
+            }
+            else {
+                $tests_ignored++;
+            }
+            next;
+        }
+
+        if ($result->{result} ne 'fail') {
+            say $rfh "## Some lines could not be parsed: ";
+            $result->{result} = 'fail';
+            $self->{result}   = 'fail';
+        }
+        print $rfh $line;
+
+    }
+
+    say $rfh "\nUsing " . @tests . " test commands.";
+    print $rfh "Ignoring $tests_ignored test commands." if $tests_ignored > 0;
+    if (@tests < 1) {
+        $result->{result} = 'fail';
+        $self->{result}   = 'fail';
+    }
+
+    $self->commit_result($result, $rfh);
+
+    return @tests;
+}
+
+sub parse_openposix_runfile {
+    my ($self, $cmd_pattern) = @_;
+    my $cmd_file_text = script_output('cat ~/openposix_test_list.txt');
+
+    my ($result, $rfh) = $self->start_result('runfile', 'parse openposix runfile');
+    say $rfh 'Parsing Openposix runfile for tests which match ' . $cmd_pattern;
+
+    my @tests = ();
+    my @lines = split qr/[\n\r\f]+/, $cmd_file_text;
+    for my $line (@lines) {
+        if ($line =~ m/$cmd_pattern/) {
+            push @tests,
+              {
+                name    => basename($line, '.run-test'),
+                command => $line
+              };
+        }
+    }
+
+    print $rfh 'Using ' . @tests . ' of ' . @lines . ' Openposix test cases';
+    if (@tests < 1) {
+        $result->{result} = 'fail';
+        $self->{result}   = 'fail';
+    }
+
+    $self->commit_result($result, $rfh);
+
+    return @tests;
+}
+
+sub parse_ltp_log {
+    my ($self, $test_log, $fin_msg, $fh) = @_;
+    my $ignored_lines = 0;
+    my ($tconf, $tfail) = (0, 0);
+
+    for (split(qr/\n/, $test_log)) {
+        if (m/^\s+$/) {
+            next;
+        }
+        elsif (m/^(\w+)\s+(\d+)\s+(\w+)\s+:\s+(.*)$/) {
+            if ($3 eq 'TFAIL' || $3 eq 'TBROK') {
+                $tfail = 1;
+                print $fh $_;
+            }
+            elsif ($3 eq 'TCONF') {
+                $tconf = 1;
+                print $fh $_;
+            }
+        }
+        elsif (m/$fin_msg(\d+)/) {
+            if ($1 == 0 && ($tfail || $tconf)) {
+                say $fh 'TEST EXIT CODE IS ZERO, YET TFAIL, TCONF OR TBROK WAS SEEN!';
+                $tfail = 1;
+            }
+            elsif ($1 == 32 && $tfail) {
+                say $fh 'TEST EXIT CODE IS 32 (TCONF), YET TFAIL OR TBROK WAS SEEN!';
+            }
+            elsif ($1 == 32) {
+                say $fh 'Test process returned TCONF (32).';
+                $tconf = 1;
+            }
+            elsif ($1 > 0) {
```

Comment:
I would be extra paranoid and wrote `$1 != 0` here.

Otherwise this version looks fine.